### PR TITLE
feat(scan): execute all checks if no checks are provided

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

If no checks are provided when the scan is initialized, all checks for the provider will be executed. Threat detection checks will also be discarded.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/scan/scan.py [hunk 1]
```diff
@@ -19,7 +20,7 @@ class Scan:
     _progress: float = 0.0
     _findings: list = []
 
-    def __init__(self, provider: Provider, checks_to_execute: list[str]):
+    def __init__(self, provider: Provider, checks_to_execute: list[str] = []):
```
Comment (jfagoagas) [inline prowler/lib/scan/scan.py]:
```suggestion
    def __init__(self, provider: Provider, checks_to_execute: list[str] = None):
```

rationale -> https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

File: scan.py
Comment (jfagoagas) [inline scan.py:1]: Watch out this file, do you want to push it?

File: prowler/lib/scan/scan.py [hunk 2]
```diff
@@ -29,11 +30,31 @@ def __init__(self, provider: Provider, checks_to_execute: list[str]):
         """
         self._provider = provider
         # Remove duplicated checks and sort them
-        self._checks_to_execute = sorted(list(set(checks_to_execute)))
+        self._checks_to_execute = (
```
Comment (jfagoagas) [inline prowler/lib/scan/scan.py:33]: Why are you doing this here? I mean the `check[0]`.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/scan/scan.py (hunk 2 above)
Comment (pedrooot) [reply prowler/lib/scan/scan.py:33]: In this case, `recover_checks_from_provider` returns a tuple with the check name and the check path, as I'm just storing the check name I take the first value from the tuple.

File: prowler/lib/scan/scan.py (hunk 1 above)
Comment (pedrooot) [reply prowler/lib/scan/scan.py]: Ok! Thanks

File: scan.py
Comment (pedrooot) [reply scan.py:1]: My bad 🤦🏼

File: prowler/lib/scan/scan.py (hunk 2 above)
Comment (jfagoagas) [reply prowler/lib/scan/scan.py:33]: Good catch, I forgot about it!

**Review by jfagoagas — APPROVED** [review]
🚀

**Pull request merged**